src/NSwag.AspNetCore/OpenApiExtensions.cs

```diff
@@ -0,0 +1,239 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="OpenApiExtensions.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Reflection;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.ApiDescription;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.FileProviders;
+using NSwag.AspNetCore;
+using NSwag.AspNetCore.DependencyInjection;
+using NSwag.AspNetCore.Middlewares;
+using NSwag.SwaggerGeneration.AspNetCore;
+
+namespace Microsoft.Extensions.DependencyInjection
```

Comment:
The namespace here is very important. Users should be able to do mainline scenarios as soon as they add the NSwag.AspNetCore package to their project. No need to add a `using NSwag.*` statement in their `Startup` class.

Reply:
Without a call to AddSwagger() the interface will not be registered and thus the whole thing wont work, right?

Reply:
That's a different concern. Everything fails nice and loudly when `AddSwagger()` is not called but the `Use*(...)` methods are.